testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueries.java

```diff
@@ -303,7 +304,13 @@ public void testShowSchemasLikeWithEscape()
             Set<Object> allSchemas = computeActual("SHOW SCHEMAS").getOnlyColumnAsSet();
             assertEquals(allSchemas, computeActual("SHOW SCHEMAS LIKE '%_%'").getOnlyColumnAsSet());
             Set<Object> result = computeActual("SHOW SCHEMAS LIKE '%$_%' ESCAPE '$'").getOnlyColumnAsSet();
-            assertThat(result).isSubsetOf(allSchemas);
+            boolean schemasWithoutUnderscoreExist = allSchemas.stream().anyMatch(schema -> ((String) schema).contains("_"));
+            if (schemasWithoutUnderscoreExist) {
+                assertNotEquals(allSchemas, result);
+            }
+            else {
+                assertThat(result).isSubsetOf(allSchemas);
```

Comment:
except for concurrency, `assertThat(result).isSubsetOf(allSchemas)`  can be asserted unconditionally, and only `assertNotEquals` needs a guard. this way, we would have assertion that `result` is strict subset of `allSchemas`, which we lacked before.

Reply:
I'll need to verify but I think `isSubsetOf` isn't strict subset. i.e. `a b c` is considered subset of `a b c`. This means if we ever had a bug in `SHOW SCHEMAS LIKE ... ESCAPE` then it can get masked because both SHOW SCHEMAS with and without ESCAPE would return same set of schemas.

That's why `assertNotEquals` is the stronger assertion.

Reply:
> That's why assertNotEquals is the stronger assertion.

i didn't mean to run isSubsetOf instead of assertNotEquals, but additionally to.